google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/FieldValue.java

```diff
@@ -229,6 +229,19 @@ public BigDecimal getNumericValue() {
     return new BigDecimal(getStringValue());
   }
 
+  /**
+   * Returns this field's value as a {@link Range}. This method should only be used * if the
+   * corresponding field has {@link LegacySQLTypeName#RANGE} type.
+   *
+   * @throws ClassCastException if the field is not a primitive type
+   * @throws IllegalArgumentException if the field's value could not be converted to {@link Range}
+   * @throws NullPointerException if {@link #isNull()} returns {@code true}
+   */
+  @SuppressWarnings("unchecked")
+  public Range getRangeValue() {
```

Comment:
How do you propagate the RangeElementType from the schema when constructing the range?  I _think_ range.of() should probably be a two-arg constructor (the inner type and string api representation).

Reply:
Updated to propagate RangeElementType when schema is available. Provides best effort when schema is not available. Also, updated Range to use FieldElementType instead of StandardSQLTypeName to be more consistent with QueryParameterValue.